### PR TITLE
[Maintenance] refactor maintenance

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
deprecate old maintenance job and introduce new tagged services for identifiers. also extract all current core maintenance tasks into separate maintenance files and remove Business Logic from models.

The goals of this PR are:
 - remove business logic from models
 - make it easier to define new maintenance tasks
 - remove as many `\Pimcore::getContainer` as possible involved in maintenance tasks
 - BC

TODO:
 - [x] docs

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
docs are added -> turned out there weren't any docs at all for maintenance tasks
also changed the ecommerce framework tasks

**Comment by dpfaffenbauer** [comment]
will check later this week

**Comment by dpfaffenbauer** [comment]
@diveshp3 Version Cleanup Task is fixed

File: bundles/CoreBundle/Command/MaintenanceCommand.php [hunk 1]
```diff
@@ -82,32 +101,33 @@ protected function execute(InputInterface $input, OutputInterface $output)
         $manager->setExcludedJobs($excludedJobs);
         $manager->setForce((bool) $input->getOption('force'));
 
-        // register scheduled tasks
-        $manager->registerJob(Job::fromMethodCall('scheduledtasks', new Schedule\Task\Executor(), 'execute'));
-        $manager->registerJob(Job::fromMethodCall('logmaintenance', new \Pimcore\Log\Maintenance(), 'mail'));
-        $manager->registerJob(Job::fromMethodCall('cleanuplogfiles', new \Pimcore\Log\Maintenance(), 'cleanupLogFiles'));
-        $manager->registerJob(Job::fromMethodCall('httperrorlog', new \Pimcore\Log\Maintenance(), 'httpErrorLogCleanup'));
-        $manager->registerJob(Job::fromMethodCall('checkErrorLogsDb', new \Pimcore\Log\Maintenance(), 'checkErrorLogsDb'));
```
Comment (dvesh3) [inline bundles/CoreBundle/Command/MaintenanceCommand.php]: service missing for checkErrorLogsDb

Comment (dpfaffenbauer) [reply bundles/CoreBundle/Command/MaintenanceCommand.php]: added

**Comment by brusch** [comment]
Thanks a lot @dpfaffenbauer !

**Pull request merged**